### PR TITLE
Update aws-sdk to 1

### PR DESCRIPTION
**Pull request by ecpullen**
*Issue #, if available:*

*Description of changes:*
Updates the AWS SDK for rust to `1`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -419,52 +439,53 @@ dependencies = [
...
+checksum = "feb5b8c7a86d4b6399169670723b7e6f21a39fc833a30f5c5a2f997608178129"
 dependencies = [
  "aws-smithy-types",
  "urlencoding",
 ]
 
 [[package]]
 name = "aws-smithy-runtime"
-version = "0.57.2"
+version = "1.1.0"
```
Comment (jpculp) [inline Cargo.lock:461]: Can this be `1.0.3` to avoid pulling in two versions of `aws-smithy-http`?

File: tough-kms/Cargo.toml [hunk 2]
```diff
@@ -16,16 +16,15 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-kms/rustls"]
...
 snafu = { version = "0.7", features = ["backtraces-impl-backtrace-crate"] }
 tokio = { version = "1", features = ["fs", "io-util", "time", "macros", "rt-multi-thread"] }
 pem = "3"
 
 [dev-dependencies]
-aws-smithy-runtime = {version = "0.57", features = ["test-util"]}
-aws-smithy-http = "0.57"
-aws-smithy-types = "0.57"
+aws-smithy-runtime = {version = "1.0", features = ["test-util"]}
+aws-smithy-types = "1.1"
```
Comment (jpculp) [inline tough-kms/Cargo.toml]: Should we make types "1.0" too and make sure `aws-smithy-runtime-api` and `aws-smithy-async` are also 1.0.3? (not sure what happens if we have mixed versions)

File: Cargo.lock (hunk 1 above)
Comment (webern) [reply Cargo.lock:461]: Only one major version will be allowed anyway, so a distinction between "1.1" and "1.0" is not meaningful (the lock file will not be consulted when using a lib from an external project). Not sure if I'm missing something.

File: tough-kms/Cargo.toml (hunk 2 above)
Comment (webern) [reply tough-kms/Cargo.toml]: We should make these say `"1"`, not `"1.1"`.

File: tough-kms/Cargo.toml [hunk 3]
```diff
@@ -16,16 +16,15 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-kms/rustls"]
 [dependencies]
 tough = { version = "0.16", path = "../tough", features = ["http"] }
 ring = { version = "0.17", features = ["std"] }
-aws-sdk-kms = "0.36"
-aws-config = "0.57"
+aws-sdk-kms = { version = "1.4", features = ["behavior-version-latest"] }
```
Comment (webern) [inline tough-kms/Cargo.toml]:
```suggestion
aws-sdk-kms = { version = "1", features = ["behavior-version-latest"] }
```

File: tough-kms/Cargo.toml [hunk 4]
```diff
@@ -16,16 +16,15 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-kms/rustls"]
...
+aws-config = { version = "1.0", features = ["behavior-version-latest"] }
```
Comment (webern) [inline tough-kms/Cargo.toml]:
```suggestion
aws-config = { version = "1", features = ["behavior-version-latest"] }
```

File: tough-kms/Cargo.toml [hunk 5]
```diff
@@ -16,16 +16,15 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-kms/rustls"]
...
 snafu = { version = "0.7", features = ["backtraces-impl-backtrace-crate"] }
 tokio = { version = "1", features = ["fs", "io-util", "time", "macros", "rt-multi-thread"] }
 pem = "3"
 
 [dev-dependencies]
-aws-smithy-runtime = {version = "0.57", features = ["test-util"]}
-aws-smithy-http = "0.57"
-aws-smithy-types = "0.57"
+aws-smithy-runtime = {version = "1.0", features = ["test-util"]}
```
Comment (webern) [inline tough-kms/Cargo.toml]:
```suggestion
aws-smithy-runtime = {version = "1", features = ["test-util"]}
```

File: tough-kms/Cargo.toml (hunk 2 above)
Comment (webern) [inline tough-kms/Cargo.toml]:
```suggestion
aws-smithy-types = "1"
```

File: tough-ssm/Cargo.toml [hunk 6]
```diff
@@ -15,7 +15,7 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-ssm/rustls"]
 
 [dependencies]
 tough = { version = "0.16", path = "../tough", features = ["http"] }
-aws-sdk-ssm = "0.36"
-aws-config = "0.57"
+aws-sdk-ssm = "1.4"
```
Comment (webern) [inline tough-ssm/Cargo.toml]:
```suggestion
aws-sdk-ssm = "1"
```

File: tough-ssm/Cargo.toml [hunk 7]
```diff
@@ -15,7 +15,7 @@ aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-ssm/rustls"]
...
+aws-config = { version = "1.0", features = ["behavior-version-latest"] }
```
Comment (webern) [inline tough-ssm/Cargo.toml]:
```suggestion
aws-config = { version = "1", features = ["behavior-version-latest"] }
```

File: tuftool/Cargo.toml [hunk 8]
```diff
@@ -15,9 +15,9 @@ aws-sdk-rust = ["aws-sdk-rust-rustls"]
 aws-sdk-rust-rustls = ["aws-config/rustls", "aws-sdk-ssm/rustls", "aws-sdk-kms/rustls",]
 
 [dependencies]
-aws-config = "0.57"
-aws-sdk-kms = "0.36"
-aws-sdk-ssm = "0.36"
+aws-config = { version = "1.0", features = ["behavior-version-latest"] }
+aws-sdk-kms = "1.4"
+aws-sdk-ssm = "1.4"
```
Comment (webern) [inline tuftool/Cargo.toml]:
```suggestion
aws-config = { version = "1", features = ["behavior-version-latest"] }
aws-sdk-kms = "1"
aws-sdk-ssm = "1"
```

**Comment by ecpullen** [comment]
^ Updated aws-sdk dependencies to `1` and ran 
```
cargo update -p aws-config -p aws-sdk-kms -p aws-sdk-ssm
```
to update `Cargo.lock`

**Review by jpculp — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Comment by webern** [comment]
Superseded by #751

**Pull request closed without merging**